non/cats
# Set up basic law-checking.

This commit does a lot of things:

 1. Sets up Discipline laws for Invariant through Applicative
 2. Uses ScalaTest to verify the laws for some instances
 3. Fixes a weird issue with Bijection's compose method
 4. Introduces ArbitraryK[F[_]] for generating Arbitrary[F[A]]s.
 5. Starts integrating algebra's std instances into cats.std.

There are still a lot of issues that need fixing:

 1. We should actually remove Bijection.
 2. Ideally we'd move from ScalaTest to specs2 or similar.
 3. Full algebra.std integration once its instances are more complete.
 4. Determine how many concrete A, B, and C types we want to test for.
 5. Add all the laws!
 6. Ensure that we're exporting the laws in a useful way.

Whew!
